eogrow/core/area/custom_grid.py

```diff
@@ -81,3 +81,46 @@ def _construct_file_path(self, *, prefix: str, suffix: str = "gpkg", **_: Any) -
         filename = filename.replace(" ", "_")
 
         return fs.path.combine(self.storage.get_cache_folder(), f"{filename}.{suffix}")
+
+
+class NewCustomGridAreaManager(BaseAreaManager):
+    """Area manager that works with a pre-defined grid of EOPatches"""
+
+    class Schema(BaseAreaManager.Schema):
+        grid_folder_key: str = Field("input_data", description="Which folder the grid file is in.")
+        grid_filename: Path = Field(
+            description=(
+                "A Geopackage with a collection of bounding boxes and attributes that will define EOPatches. In"
+                " case bounding boxes are in multiple CRS then each Geopackage layer should contain bounding boxes"
+                " from one CRS."
+            ),
+            regex=r"^.+\..+$",
+        )
+        name_column: str = Field(description="Name of the column containing EOPatch names.")
+
+    config: Schema
+
+    def _create_grid(self) -> Dict[CRS, gpd.GeoDataFrame]:
+        grid_path = fs.path.combine(self.storage.get_folder(self.config.grid_folder_key), self.config.grid_filename)
+        grid = self._load_grid(grid_path)
+
+        for crs, crs_gdf in grid.items():
+            # Correct name of eoptach-name-column, drop all non-significant ones
+            names = crs_gdf[self.config.name_column]
+            grid[crs] = gpd.GeoDataFrame(geometry=crs_gdf.geometry, data={self.NAME_COLUMN: names}, crs=crs_gdf.crs)
+
+        return grid
+
+    def get_area_geometry(self, *, crs: CRS = CRS.WGS84) -> Geometry:
+        all_grid_gdfs = list(self.get_grid().values())
+        area_df = concat_gdf(all_grid_gdfs, reproject_crs=crs)
+
+        LOGGER.info("Calculating a unary union of the area geometries")
+        area_shape = shapely.ops.unary_union(area_df.geometry)
+        return Geometry(area_shape, crs)
```

Comment:
@mlubej  now that i think about this, the 'area geometry' is usually the buffered and simplified AOI, but in the case of the custom grid I just took all the elements of the grid. Do you think it would be more correct to take the union of all *geometries* (and not the bboxes that they define) of the custom grid? Usually a custom grid will contain preciselly BBox geometries, but not necessarily.

Reply:
If the custom grid is pre-defined, do you even have access to the original geometries? I imagine if the user just wants to proved the new grid, they are not interested in the original geometries anymore (maybe)

Reply:
I actually think it already does what i wanted it to do, so i'm resolving this :see_no_evil: